File: 283-move-zeroes/move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        
        int nextindex=0;

        //Move all zero to the front
        for(int i=0;i<nums.size();i++){
            if(nums[i]!=0){
                nums[nextindex++]=nums[i];
            }
        }

        for(int i=nextindex;nextindex<nums.size();nextindex++){
            nums[i++]=0;
        }
    }
};

File: 283-move-zeroes/move-zeroes.py
class Solution:
    def moveZeroes(self, nums: list[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        idx=0
        for i,x in enumerate(nums):
            if x!=0:
                nums[idx],nums[i]=nums[i],nums[idx]
                idx+=1
        return idx        